internal/runners/branch/tree.go

```diff
@@ -17,108 +16,87 @@ type branchNode struct {
-		}
-		rootBranches = append(rootBranches, branch)
-	}
-	return rootBranches
-}
+func branchListing(branches mono_models.Branches, localBranch string) []string {
+	sort.Slice(branches, func(i, j int) bool {
+		return branches[i].Label < branches[j].Label
+	})
 
-func containsBranch(id *strfmt.UUID, branches mono_models.Branches) bool {
+	var branchNames []string
 	for _, branch := range branches {
-		if branch.BranchID.String() == id.String() {
-			return true
-		}
+		branchName := applyFormatting(branch.Label, localBranch)
+		branchNames = append(branchNames, branchName)
 	}
-	return false
+	return branchNames
 }
 
-func getChildren(branch *mono_models.Branch, branches mono_models.Branches) tree {
-	children := make(tree)
-	if branch == nil {
-		return children
+func branchTree(branches mono_models.Branches, localBranch string) string {
+	tree := make(tree)
+	for _, branch := range model.GetRootBranches(branches) {
+		tree[branchNode{branch.Label, branch.BranchID.String()}] = buildBranchTree(branch, branches)
 	}
 
-	for _, b := range branches {
-		if b.Tracks == nil {
+	var levelsCompleted []int
+	rootBranches := model.GetRootBranches(branches)
+	return treeString(tree, rootBranches, localBranch, 0, levelsCompleted)
+}
+
+func buildBranchTree(currentBranch *mono_models.Branch, branches mono_models.Branches) tree {
+	children := getChildren(currentBranch, branches)
+	for _, branch := range branches {
+		// Discard any branches without tracking information as we are only interested
+		// in child branches of the current branch
+		if branch.Tracks == nil {
 			continue
 		}
-		if b.Tracks.String() == branch.BranchID.String() {
-			children[branchNode{b.Label, b.BranchID.String()}] = make(tree)
+
+		// Check that this branch is a child branch and recursively build its tree
+		node := branchNode{branch.Label, branch.BranchID.String()}
+		if _, ok := children[node]; ok {
+			children[node] = buildBranchTree(branch, branches)
 		}
 	}
 	return children
 }
 
-func (bt *BranchTree) AddBranchFormatting(formatting string) {
-	bt.branchFormatting = formatting
-}
-
-func (bt *BranchTree) AddLocalBranch(branch string) {
-	bt.localBranch = branch
-}
-
-func (bt *BranchTree) AddLocalBranchFormatting(formatting string) {
-	bt.localBranchFormatting = formatting
-}
+func getChildren(branch *mono_models.Branch, branches mono_models.Branches) tree {
+	childTree := make(tree)
+	children := model.GetBranchChildren(branch, branches)
 
-func (bt *BranchTree) String() string {
-	w := new(bytes.Buffer)
-	var levelsCompleted []int
-	bt.print(w, bt.tree, 0, levelsCompleted)
-	return w.String()
+	for _, child := range children {
+		childTree[branchNode{child.Label, child.BranchID.String()}] = make(tree)
+	}
+	return childTree
 }
 
-func (bt *BranchTree) print(w io.Writer, currentTree tree, currentLevel int, levelsCompleted []int) {
+func treeString(currentTree tree, rootBranches mono_models.Branches, localBranch string, currentLevel int, levelsCompleted []int) string {
```

Comment:
`treeString` and `nodeString` below feel like they should be `.String()` methods on a struct. And as said before at the moment they do way too much.